### ISSUE TITLE
Images do not nest in html

### ISSUE THREAD

**Issue by daslu** [nextjournal/clerk#118]
The following example:
```clj
(ns images
  (:require [nextjournal.clerk :as clerk]))

(import javax.imageio.ImageIO
        java.net.URL)

(clerk/html
 [:div
  [:h1 "image1"]
  (ImageIO/read (URL. "https://images.unsplash.com/photo-1532879311112-62b7188d28ce?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8"))
  [:h1 "image2"]
  (ImageIO/read (URL. "https://images.unsplash.com/photo-1532879311112-62b7188d28ce?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8"))])
```
results in the following:
![clerk-example](https://user-images.githubusercontent.com/5673102/159462709-68839b8b-9155-4834-be6d-0b7dc33526c8.png)

Thanks!

**Comment by jackrusher** [comment]
I have no idea what you had expected to happen in this case.

**Comment by daslu** [comment]
Thanks,

Here is the Slack discussion for reference:
https://clojurians.slack.com/archives/C035GRLJEP8/p1647906226495609

What I was wondering about was how to embed `BufferedImage` values in a Hiccup block.

**Comment by jackrusher** [comment]
The linked Slack thread doesn't make clear what output you might have expected given the semantics of `clerk/html`, which in turn leaves me unsure how to help you.

**Comment by daslu** [comment]
I might have expected to see the corresponding images inside the div, since typically, Clerk viewer types nest.

**Comment by jackrusher** [comment]
Let me try to refine my question a bit: if you are using the `html` viewer to build a hiccup representation of some HTML you would like to show the user, what would it mean to dump the binary data of an image in the middle?

I would expect something like `[:img {:src (url-for (ImageIO/read ...))}]` in this place so that you have control over the CSS applied to the element, can supply click handlers, and so on. (Note that this currently doesn't work! But to come up with a good solution that gives the user full power and no magic, we need to consider what _should_ happen.)

**Comment by daslu** [comment]
Thanks, yes, I'm not sure about the desired API.

So, maybe the ticket title should have been: "What is the recommended practice for showing a few images in a div?".

A few thoughts:
* Currently, a top-level form returning a `BufferedImage` just renders as an image in the Clerk notebook, without asking the user to care about serving it through a URL, etc.
* This is quite useful. Example use cases:
  * The user is doing some image processing and wants to just see the resulting image.
  * The user is getting an image from a plotting library (e.g., [cljplot](github.com/generateme/cljplot/)) and just wants to see it.
* I think that Clerk tends to render things (e.g., Vega plots) the same when nested inside a div and when not.
* I think this kind of consistency is nice. That is probably what made me assume some desired behavior here.

**Comment by jackrusher** [comment]
I agree that serving `BufferedImage`s as visible images is useful, that's why we made that feature! In this case, it's not clear what we want to do inside `html` blocks, so there's going to be some design work around what feels right. :)

**Comment by mk** [comment]
@daslu it took a while but it's now working with a8c5102dc21322e1804af334308d217987e94568:
<img width="938" alt="Screen Shot 2023-02-21 at 15 43 58" src="https://user-images.githubusercontent.com/1187/220376193-f358084c-bd55-40bd-aafc-a42f71120753.png">
